coderd/workspaceapps/db.go

```diff
@@ -102,11 +85,13 @@ func (p *DBTokenProvider) CreateToken(ctx context.Context, rw http.ResponseWrite
 		OAuth2Configs:               p.OAuth2Configs,
 		RedirectToLogin:             false,
 		DisableSessionExpiryRefresh: p.DeploymentValues.DisableSessionExpiryRefresh.Value(),
-		// Optional is true to allow for public apps. If an authorization check
-		// fails and the user is not authenticated, they will be redirected to
-		// the login page using code below (not the redirect from the
-		// middleware itself).
+		// Optional is true to allow for public apps. If the authorization check
+		// (later on) fails and the user is not authenticated, they will be
+		// redirected to the login page or app auth endpoint using code below.
```

Comment:
I'm just assuming there are tests for the permissions of this. Pls verify, it seems spooky that toggling this could cause unintended and maybe non-obvious side effects.

Reply:
Yes, this is probably the most throughly tested auth code in the whole product. There's an entire matrix of permissions and what should happen. See the apptest package

Reply:
This also has already been the case for many months.